### PR TITLE
Index based

### PR DESCRIPTION
**Pull request by sjboeing**
Hi @matt-frey. Here is the work that stores the contributions to the grid in case support points correspond to the same index. Let me know if you can check it for the issue you spotted before.

### CODE REVIEW

**Comment by sjboeing** [comment]
My changes to the moist setup are based on this one, so would be best to merge this one first if all is OK.

**Comment by sjboeing** [comment]
One thing that originally did not work was OpenMP (I had to fix line length).

**Comment by matt-frey** [comment]
I ran with OpenMP. Yes, I remember a line length issue.

**Comment by matt-frey** [comment]
@sjboeing Have you tested that you still get the same results?

File: src/3d/parcels/parcel_interpl.f90 [hunk 1]
```diff
@@ -326,74 +375,98 @@ pure subroutine trilinear(pos, ii, jj, kk, ww)
...
+            double precision, intent(inout)  :: ww(ngp)
+            integer, intent(in)              :: i_lower, j_lower, k_lower
+            double precision                 :: xyz(3)
+            xyz = (pos - lower) * dxi
+            call trilinear_weights_inner(xyz, i_lower, j_lower, k_lower, ww)
+
+        end subroutine trilinear_weights_add
+
+
+        pure subroutine trilinear_weights_inner(xyz, i_lower, j_lower, k_lower, ww)
```
Comment (matt-frey) [inline src/3d/parcels/parcel_interpl.f90:439]: Can we maybe rename this to `get_weights` or similar? I would also change `i_lower` etc. to just `i`.

File: src/3d/parcels/parcel_interpl.f90 [hunk 2]
```diff
@@ -326,74 +375,98 @@ pure subroutine trilinear(pos, ii, jj, kk, ww)
...
+
+            pz = xyz(3) - dble(k_lower)
+            pzc = one - pz
+
+            w00 = pyc * pxc
+            w10 = pyc * px
+            w01 = py * pxc
+            w11 = py * px
+
+            ww(1) = ww(1)+pzc * w00
```
Comment (matt-frey) [inline src/3d/parcels/parcel_interpl.f90:461]: Please add spaces before and after `+`. Thanks.

**Review by matt-frey — CHANGES_REQUESTED** [review]
I have two remarks.

**Comment by sjboeing** [comment]
Just revised the subroutines. Let me know if it is ready to merge.

File: src/3d/parcels/parcel_interpl.f90 (hunk 1 above)
Comment (sjboeing) [reply src/3d/parcels/parcel_interpl.f90]: Changed it to get_weights, which is indeed more explicit, with indices i, j, k (also used this naming in trilinear_weights_add now).

File: src/3d/parcels/parcel_interpl.f90 (hunk 2 above)
Comment (sjboeing) [reply src/3d/parcels/parcel_interpl.f90]: Done this.

Comment (matt-frey) [reply src/3d/parcels/parcel_interpl.f90]: Thanks!

**Review by matt-frey — APPROVED** [review]
@sjboeing I fixed a few more white spaces (with https://github.com/matt-frey/epic/pull/286/commits/6f8c49e8c189d6423ca6606eda181a7e65115e07). Thanks a lot for working on this!

**Pull request merged**